components/overflow-group/overflow-group-mixin.js

```diff
@@ -0,0 +1,349 @@
+				isForcedOverflowing = true;
+				itemLayout.isChomped = true;
+				itemLayout.trigger = 'force-hide';
+				continue;
+			}
+
+			// chomp or unchomp based on space available, and we've already handled min/max above
+			if (!isSoftOverflowing && showing.width + itemLayout.width < this._availableWidth) {
+				showing.width += itemLayout.width;
+				showing.count += 1;
+				itemLayout.isChomped = false;
+				itemLayout.trigger = 'soft-show';
+
+			} else {
+				isSoftOverflowing = true;
+				itemLayout.isChomped = true;
+				itemLayout.trigger = 'soft-hide';
+
+			}
+
+		}
+		// if there is at least one showing and no more to be hidden, enable collapsing more button to [...]
+		this.overflowMenuHidden = this._itemLayouts.length === showing.count;
+		if (!this.overflowMenuHidden && (isSoftOverflowing || isForcedOverflowing)) {
+			for (let j = this._itemLayouts.length; j--;) {
+				if (showing.width + this._overflowMenuWidth < this._availableWidth) {
+					break;
+				}
+				const itemLayoutOverflowing = this._itemLayouts[j];
+				if (itemLayoutOverflowing.trigger !== 'soft-show') {
+					continue;
+				}
+				showing.width -= itemLayoutOverflowing.width;
+				showing.count -= 1;
+				isSoftOverflowing = true;
+				itemLayoutOverflowing.trigger = 'soft-hide';
+				itemLayoutOverflowing.isChomped = true;
+			}
+		}
+		const overflowDropdownOverflowing = (showing.width + this._overflowMenuWidth >= this._availableWidth);
+		const swapToMiniButton = overflowDropdownOverflowing && !this.overflowMenuHidden;
+
+		this.mini = this.openerType === OPENER_TYPE.ICON || swapToMiniButton;
+		this.chompIndex = this.overflowMenuHidden ? null : showing.count;
+
+		/** Dispatched when there is an update performed to the overflow group */
+		this.dispatchEvent(new CustomEvent('d2l-overflow-group-updated', { composed: false, bubbles: true }));
+	}
+
+	_getItemLayouts(filteredNodes) {
+		const items = filteredNodes.map((node) => {
+			const computedStyles = window.getComputedStyle(node);
+
+			return {
+				type: node.tagName.toLowerCase(),
+				isChomped: false,
+				isHidden: computedStyles.display === 'none',
+				width: Math.ceil(parseFloat(computedStyles.width) || 0)
+					+ parseInt(computedStyles.marginRight) || 0
+					+ parseInt(computedStyles.marginLeft) || 0,
+				node: node
+			};
+		});
+
+		return items.filter(({ isHidden }) => !isHidden);
+	}
+
+	async _getItems() {
+		// get the items from the button slot
+		this._slotItems = await this._getSlotItems();
+		// convert them to layout items (calculate widths)
+		this._itemLayouts = this._getItemLayouts(this._slotItems);
+		// convert to dropdown items (for overflow menu)
+		this.dropdownItems = this._slotItems.map((node) => this.convertToOverflowItem(node));
+	}
+
+	async _getSlotItems() {
+		const filteredNodes = await filterAsync(this._buttonSlot.assignedNodes({ flatten: true }), async(node) => {
+			if (node.nodeType !== Node.ELEMENT_NODE) return false;
+			await node.updateComplete;
```

Comment:
This seems to assume that the node has an `updateComplete` property that returns a promise.  What happens if the node is a native element such as a `button` or a `template`?

Reply:
So at least with `button` it just seems to skip over and still return the element as expected. However if you console log `node.updateComplete` it is undefined so best to avoid that altogether. Is just adding an `if (node.updateComplete) await node.updateComplete` fine or is there a best practice for this?

Reply:
I did a quick test running `await undefined` and the browser just seems to ignore it, so sounds like what you have is safe.